src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -53,8 +53,28 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
 
     auto sliding_window = v0::Constant::create(element::i32, Shape{}, {0});  // sliding_window
 
+    auto has_parameter = [=](const std::shared_ptr<ov::Model>& model,
+                             const std::string& name) -> std::shared_ptr<v0::Parameter> {
+        for (auto& param : model->inputs()) {
+            const auto& names = param.get_names();
+            if (names.find(name) != names.end()) {
+                if (auto casted_param = std::dynamic_pointer_cast<v0::Parameter>(param.get_node_shared_ptr())) {
+                    return casted_param;
+                } else {
+                    OPENVINO_THROW("The model is in the inconsistent state. Found input '",
+                                   name,
+                                   "', but couldn't cast it to v0::Parameter.");
+                }
```

Comment:
```suggestion
                auto casted_param = ov::as_type_ptr<v0::Paramter>(param.get_node_shared_ptr()));
                OPENVINO_ASSERT(casted_param , "The model is in the inconsistent state. Input is not paramter: ", casted_param);
                return casted_param;
```

Use `ov::as_type_ptr<>` instead of `std::dynamic_pointer_cast` for operators.

Reply:
Fixed

Reply:
Assertion not changed, it can remove number of `if`  statement.